crypto/x509/v3_purp.c

```diff
@@ -117,6 +120,16 @@ int X509_PURPOSE_get_count(void)
     return sk_X509_PURPOSE_num(xptable) + X509_PURPOSE_COUNT;
 }
 
+/* find smallest identifier not yet taken - note there might be gaps */
+int X509_PURPOSE_get_unused_id(void)
```

Comment:
One more thing to consider. If we ever decide to move the purposes table to a library context, this call will need to be replaced with an _ex() function. Perhaps we should consider adding a libctx argument, that would be currently ignored.

Reply:
Interesting point - added that parameter.